test/conformance/service_test.go

```diff
@@ -536,4 +536,51 @@ func TestReleaseService(t *testing.T) {
 	}
 }
 
+func TestMultipleNamespace(t *testing.T) {
+	t.Parallel()
+
+	altServiceNamespace := fmt.Sprintf("%s-%s", test.ServingNamespace, test.ObjectNameForTest(t))
+	serviceName := test.ObjectNameForTest(t)
+
+	defaultClients := e2e.Setup(t) // This one uses the default namespace `test.ServingNamespace`
+	altClients := e2e.SetupWithNamespace(t, altServiceNamespace)
+
+	_, err := test.CreateNamespace(altClients, altServiceNamespace)
+	if err != nil {
+		t.Error(err)
+	}
+	defer test.DeleteNamespace(altClients, altServiceNamespace)
+
+	deploy := func(clients *test.Clients, names *test.ResourceNames) *test.ResourceObjects {
+		objects, err := test.CreateRunLatestServiceReady(t, clients, names, &test.Options{})
+		if err != nil {
+			t.Fatalf("Failed to create Service %v: %v", names.Service, err)
```

Comment:
I'd return the error and fail in the main test body.

Reply:
I'm not seeing the benefit of bubbling that error up to the main test body. If I do that, may as well just put the function inline.

Reply:
Well the problem is when you're failing there the error will carry that line, which won't show the actual problem (especially that you don't print the namespace either).
I guess you can add `t.Helper()`, but returning error would permit you to do `t.Error` at the precise point, which aides with debugging.

Reply:
Fair point on the debugging. I'll inline that method and update it to print the namespace as well.